feat(sdk): upgrade kms core dependencies and update user key pair algorithm accordingly

leventdem opened the pull request:
This PR introduces an upgrade of the user decryption key pair generation algorithm. 

Initially cryptobox was used, but now ml-kem is used to enable post-quantum security.

The API changes with the new kms crate version:
- KMS signers are given with the corresponding IDs
- Direct call to client is done instead of calling a function and passing a mutable client

Others changes:
- keys are longer
- use separate json file to test ( a test generated on kms core has been used to verify behavior)

immortal-tofu reviewed (commented):
I did a quick review. I think most of the task should be moved to fhevm-backend because writing stuff in node_modules is not a good practice, especially because on `npm install` any change to `node_modules` will be removed.
Not sure if it's problematic to move these tasks (upgradeProxy, etherscanveriufy, taskTFHE, taskDeploy, ...)

leventdem committed:
chore(sdk): add domain to builder of response

Add ability to customize domain name for user decryption response
verification.

Default value is "Decryption"

leventdem committed:
chore(sdk): update deserialization tool

Use bc2wrap to deserialize user decryption response (json/hex format)

leventdem committed:
feat(sdk): use ml-kem for user decryption keys

Update new post-quantum secure algorithm for user decryption.

Separate test data in a json file

leventdem committed:
chore(sdk): update dependencies

Force specific version of alloy to corresponds to kms

leventdem committed:
feat(sdk): update key gen algorithm for signature

leventdem committed:
chore(sdk): remove insecure flag

ppoliani commented:
@leventdem [this](https://github.com/zama-ai/fhevm/blob/b7b2485161993cb7cd16d824d97d6d901879da3a/sdk/rust-sdk/src/decryption/user/response.rs#L447) test was failing when I applied a similar change to the.

The issue is that we're using the j`s_api` from kms-core which targets wasm. Currently when a JsError is created we get a panic.

> running 1 test
panicked at /wasm-bindgen-0.2.100/src/lib.rs:1062:1:
function not implemented on non-wasm32 targets
panicked at library/core/src/panicking.rs:218:5:

Should we try avoid using the wasm interface within the `gateway-sdk`?

leventdem committed:
refactor(sdk): improve builder with consistency on methods names

leventdem committed:
chore(sdk): fix clippy warnings

leventdem committed:
chore(sdk): add CI workflow for running test

leventdem committed:
chore(sdk): fix cargo fmt

leventdem commented:
>  Should we try avoid using the wasm interface within the gateway-sdk?

For now, we have to use some wasm compatible method, because initially the crate was implemented for wasm. 
A more full-rust version will be available in kms.

leventdem committed:
chore(sdk): fix clippy warnings

leventdem committed:
chore(sdk): fmt

leventdem committed:
chore(sdk): fix example
